wren-core/core/src/mdl/manifest.rs

```diff
@@ -18,6 +18,50 @@ pub struct Manifest {
     pub metrics: Vec<Arc<Metric>>,
     #[serde(default)]
     pub views: Vec<Arc<View>>,
+    #[serde(default)]
+    pub data_source: Option<DataSource>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Hash, Clone, Copy)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum DataSource {
+    #[serde(alias = "bigquery")]
+    BigQuery,
+    #[serde(alias = "clickhouse")]
+    Clickhouse,
+    #[serde(alias = "canner")]
+    Canner,
+    #[serde(alias = "trino")]
+    Trino,
+    #[serde(alias = "mssql")]
+    MSSQL,
+    #[serde(alias = "mysql")]
+    MySQL,
+    #[serde(alias = "postgres")]
+    Postgres,
+    #[serde(alias = "snowflake")]
+    Snowflake,
+    #[serde(alias = "datafusion")]
+    Datafusion,
+    #[serde(alias = "duckdb")]
+    DuckDB,
+}
+
+impl Display for DataSource {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            DataSource::BigQuery => write!(f, "BIGQUERY"),
+            DataSource::Clickhouse => write!(f, "CLICKHOUSE"),
+            DataSource::Canner => write!(f, "CANNER"),
+            DataSource::Trino => write!(f, "TRINO"),
+            DataSource::MSSQL => write!(f, "MSSQL"),
+            DataSource::MySQL => write!(f, "MYSQL"),
+            DataSource::Postgres => write!(f, "POSTGRES"),
+            DataSource::Snowflake => write!(f, "SNOWFLAKE"),
+            DataSource::Datafusion => write!(f, "DATAFUSION"),
+            DataSource::DuckDB => write!(f, "DUCKDB"),
```

Comment:
Curious, why make them uppercase?

Reply:
No, specific reason. Just use the same style as the relationship type for the enum value.

Reply:
The ibis-server and web use lowercase. Does this change will not affect the JSON format?

Reply:
I think it won't affect it. We also accept the lowercase value.

Reply:
When the ibis-server calls `ManifestExtractor` and then gets the object with an uppercase data source name, can it map to the `DataSource` enum of ibis-server?

Reply:
It depends on how to implement it. By the way, why do we need to map to the enum of ibis-server?

Reply:
It's fine for me to change them to lowercase if you think it's better. It doesn't matter actually. 🤔

Reply:
Thanks